x-pack/plugins/monitoring/public/components/shared/toolbar.tsx

```diff
@@ -12,17 +12,41 @@ import {
   EuiTitle,
   OnRefreshChangeProps,
 } from '@elastic/eui';
-import React, { useContext, useCallback } from 'react';
+import React, { useContext, useCallback, useMemo } from 'react';
 import { MonitoringTimeContainer } from '../../application/hooks/use_monitoring_time';
 import { GlobalStateContext } from '../../application/contexts/global_state_context';
 import { Legacy } from '../../legacy_shims';
-
+import { MonitoringStartServices } from '../../types';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import { UI_SETTINGS } from '../../../../../../src/plugins/data/common';
 interface MonitoringToolbarProps {
   pageTitle?: string;
   onRefresh?: () => void;
 }
 
+interface TimePickerQuickRange {
+  from: string;
+  to: string;
+  display: string;
+}
+
 export const MonitoringToolbar: React.FC<MonitoringToolbarProps> = ({ pageTitle, onRefresh }) => {
+  const { services } = useKibana<MonitoringStartServices>();
+
+  const timePickerQuickRanges = services.uiSettings.get<TimePickerQuickRange[]>(
+    UI_SETTINGS.TIMEPICKER_QUICK_RANGES
+  );
+
+  const commonlyUsedRanges = useMemo(
+    () =>
+      timePickerQuickRanges.map(({ from, to, display }) => ({
+        start: from,
+        end: to,
+        label: display,
+      })),
+    [timePickerQuickRanges]
+  );
```

Comment:
If `TIMEPICKER_QUICK_RANGES` don't change during the lifecycle of the app, they could be hoisted at module scope outside of the component... but they depend on a `useKibana` hook which must live inside of a component 🤷‍♂️

Reply:
Yep, it's a fair point. I used this chunk of code as-is from other solutions. This code (verbatim) exists in quite a few places now. We should really extract it (most likely to the `kibana_react` helpers) but that would be way out of scope for this bug fix. I can file an issue for later though (which wouldn't just affect Observability).